src/xSTIR/cSTIR/include/sirf/STIR/stir_data_containers.h

```diff
@@ -893,12 +890,24 @@ namespace sirf {
 		{
 			return false;
 		}
-
 		unsigned int items() const
 		{
 			return 1;
 		}
-        /// Write to file
+
+		std::string modality() const
+		{
+			ExamInfo ex_info = data().get_exam_info();
+			return ex_info.imaging_modality.get_name();
+		}
+		void set_modality(const char* mod)
```

Comment:
I prefer `const std::string&`. `char*` gets converted to `string`, but not the other way around. Of course, this is in many other places as well, but we can start here.

(If you need to keep `char *`, much better is `const char * const`)